lib/tasks/sandbox_data.rake

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+namespace :lead_provider do
+  desc "create seed mentors for API testing"
+  task seed_mentors: :environment do
+    logger = Logger.new($stdout)
+    logger.formatter = proc do |_severity, _datetime, _progname, msg|
+      "#{msg}\n"
+    end
+
+    providers_names = ["Capita", "Teach First", "UCL", "Best Practice Network", "Ambition", "Education Development Trust"]
+    cohort_2021 = Cohort.find_by!(start_year: "2021")
+
+    providers_names.each_with_index do |provider_name, index|
+      logger.info("Generating seed mentors for '#{provider_name}'...")
+      lead_provider = LeadProvider.find_by!(name: provider_name)
+      school = create_school_and_associations(lead_provider, cohort_2021, index)
+      generate_mentors(lead_provider, school, cohort_2021, logger)
+    end
+  end
+end
+
+def generate_mentors(lead_provider, school, cohort, logger)
+  existing_count = User.mentors_for_lead_provider(lead_provider).count
+  10.times do
+    mentor = User.create!(full_name: Faker::Name.name, email: Faker::Internet.email)
+    MentorProfile.create!(user: mentor, school: school, cohort: cohort)
+    logger.info(" Mentor with user_id #{mentor.id} generated")
+  end
+  new_count = User.mentors_for_lead_provider(lead_provider).count
+  logger.info(" Before: #{existing_count}, after: #{new_count}")
+end
+
+def create_school_and_associations(lead_provider, cohort, index)
+  school = School.find_by!(
```

Comment:
I think it's gonna fail if school doesn't exist and the previous sandbox seed file hasn't been run before

Reply:
It's only for the sandbox which it's been run on, do we care?

Reply:
Oh actually you can't run `db:reset` then this, okay I can see why we need it to find_or_create now. Fixing...
Thanks

Reply:
Is the fix `find_or_create_by!`? If so, then maybe we don't really care, but the fix is so small and simple that, well, why not?

Reply:
Done the schools etc, but this has flushed out a similar issue with the UCL provider, I need to look a bit closer at this....